utils/anautilities.py

```diff
@@ -442,6 +442,59 @@ def filePathExists(searchPath, subPath=None, debug=False):
             print "Found %s"%s(testPath)
         return True
 
+#========================================================================================
+# Find Inflection Point /////////////////////////////////////////////////////////////////
+#----------------------------------------------------------------------------------------
+# Author(s): Brendan Regnery, Brian Dorney, Jared Sturdy ////////////////////////////////
+#----------------------------------------------------------------------------------------
+
+def findInflectionPts(graph):
+    '''
+    Find and return the "knee", inflection point, of the SBit Rate scan
+    graph should be a TGraph or TH1F of the SBit Rate scan for 1 vfat 
+    '''
+
+
+    import ROOT as root
+    import numpy as np
+    from itertools import groupby
+
+    # make TH1F into TGraph
+    if type(graph) == root.TH1F :
+        graph = root.TGraph(graph)
+
+    # make arrays out of the x and y components
+    x = graph.GetX()
+    y = graph.GetY()
+    x = np.array(x, dtype=float)
+    y = np.array(y, dtype=float)
+
+    # Calculate the gradient of y as a function of x
+    # Documentation here: https://docs.scipy.org/doc/numpy/reference/generated/numpy.gradient.html
+    grad = np.gradient(y, x)
+
+    # Split the array when the slope changes sign
+    posGrad = [list(g) for k, g in groupby(grad, lambda x: x > 0) if k]
+    negGrad = [list(g) for k, g in groupby(grad, lambda x: x < 0) if k]
```

Comment:
What are k & g supposed to be here?  Single letter variable names (even in list comprehension statements) are generally not a good idea.

Reply:
I tend to (strongly) disagree in this case, primarily because the variables don't exist outside this single line scope, and the documentation, if needed, should be looked up in the `itertools.groupby` docs.
Here the comment already tells in a succinct way, what the lists are (well, it could explicitly state whether it stores the index of the input `grad` array, or the partitions of gradients for every sign change, but that's candy)